### PR TITLE
Only animate CGSize and CGPoint key paths additively if additive is enabled.

### PR DESCRIPTION
**Pull request by jverkoey**
Prior to this change, CGSize and CGPoint animations were being animated additively even if the additive property was disabled.

Added tests to catch regressions in the future.

### CODE REVIEW

File: tests/unit/AdditiveAnimatorTests.swift [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  override func setUp() {
+    super.setUp()
+
+    animator = MotionAnimator()
+
+    animator.additive = true
+
+    timing = MotionTiming(delay: 0,
+                          duration: 1,
+                          curve: MotionCurveMakeBezier(p1x: 0, p1y: 0, p2x: 0, p2y: 0),
```
Comment (romoore) [inline tests/unit/AdditiveAnimatorTests.swift]: I'm probably misunderstanding timing curves, but doesn't this timing curve mean "never go to the 'to' value"? If all control points are zero?

File: tests/unit/AdditiveAnimatorTests.swift [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                    + "\(animationKeys).")
+    guard let key = animationKeys.first,
+      let animation = view.layer.animation(forKey: key) as? CABasicAnimation else {
+        return
+    }
+
+    XCTAssertTrue(animation.isAdditive, "Animation is not additive when it should be.")
+  }
+
+  func testPositionKeyPathsAnimateAdditively() {
```
Comment (romoore) [inline tests/unit/AdditiveAnimatorTests.swift]: Nit/naming: test__Point__KeyPathsAnimateAdditively

File: tests/unit/AdditiveAnimatorTests.swift [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                   "Expected only one animation to be added, but the following were found: "
+                    + "\(animationKeys).")
+    guard let key = animationKeys.first,
+      let animation = view.layer.animation(forKey: key) as? CABasicAnimation else {
+        return
+    }
+
+    XCTAssertTrue(animation.isAdditive, "Animation is not additive when it should be.")
+  }
+}
```
Comment (romoore) [inline tests/unit/AdditiveAnimatorTests.swift:118]: Are CGRect not additive no matter what?

**Review by romoore — APPROVED** [review]

File: tests/unit/AdditiveAnimatorTests.swift (hunk 1 above)
Comment (jverkoey) [reply tests/unit/AdditiveAnimatorTests.swift]: There are four control points in a cubic bezier, p0 and p3 represent the anchor points — respectively anchored at 0, 0, and 1,1 — and p1 and p2 are the control points. This API accepts p1 and p2.

---

While this particular cubic bezier is odd, it's not particularly important for the purposes of this test.

File: tests/unit/AdditiveAnimatorTests.swift (hunk 2 above)
Comment (jverkoey) [reply tests/unit/AdditiveAnimatorTests.swift]: Done.

File: tests/unit/AdditiveAnimatorTests.swift (hunk 3 above)
Comment (jverkoey) [reply tests/unit/AdditiveAnimatorTests.swift:118]: Currently no. CGRect support isn't currently implemented.

Comment (jverkoey) [reply tests/unit/AdditiveAnimatorTests.swift:118]: https://github.com/material-motion/motion-animator-objc/issues/74

**Pull request merged**